src/util/triplea/MapXMLCreator/TerritoryConnectionsPanel.java

```diff
@@ -0,0 +1,256 @@
+			scalePixels = lineThickness * 4;
+			minOverlap = scalePixels * 4;
+		} catch (final NumberFormatException ex)
+		{
+		}
+		if (JOptionPane.showConfirmDialog(null, "Scale set to " + scalePixels + " pixels larger, and minimum overlap set to " + minOverlap + " pixels. \r\n"
+					+ "Do you wish to continue with this? \r\nSelect Yes to continue, Select No to override and change the size.", "Scale and Overlap Size", JOptionPane.YES_NO_OPTION) == 1)
+		{
+			final String scale = JOptionPane.showInputDialog(null,
+						"Enter the number of pixels larger each territory should become? \r\n(Normally 4x bigger than the border line width. eg: 4, or 8, etc)", "4");
+			if (scale == null)
+				return;
+			try
+			{
+				scalePixels = Integer.parseInt(scale);
+			} catch (final NumberFormatException ex)
+			{
+			}
+			final String overlap = JOptionPane.showInputDialog(null,
+						"Enter the minimum number of overlapping pixels for a connection? \r\n(Normally 16x bigger than the border line width. eg: 16, or 32, etc.)", "16");
+			if (overlap == null)
+				return;
+			try
+			{
+				minOverlap = Integer.parseInt(overlap);
+			} catch (final NumberFormatException ex)
+			{
+			}
+		}
+		
+		MapXMLHelper.s_territoyConnections.clear();
+		System.out.print("Now Scanning for Connections ... ");
+		// sort so that they are in alphabetic order (makes xml's prettier and easier to update in future)
+		final List<String> allTerritories = s_polygons == null ? new ArrayList<String>() : new ArrayList<String>(s_polygons.keySet());
+		Collections.sort(allTerritories, new AlphanumComparator());
+		final List<String> allAreas = new ArrayList<String>(territoryAreas.keySet());
+		Collections.sort(allAreas, new AlphanumComparator());
+		for (final String territory : allTerritories)
+		{
+			final LinkedHashSet<String> thisTerritoryConnections = new LinkedHashSet<String>();
+			final List<Polygon> currentPolygons = s_polygons.get(territory);
+			for (final Polygon currentPolygon : currentPolygons)
+			{
+				
+				final Shape scaledShape = ConnectionFinder.scale(currentPolygon, scalePixels);
+				
+				for (final String otherTerritory : allAreas)
+				{
+					if (otherTerritory.equals(territory))
+						continue;
+					if (thisTerritoryConnections.contains(otherTerritory))
+						continue;
+					if (MapXMLHelper.s_territoyConnections.get(otherTerritory) != null && MapXMLHelper.s_territoyConnections.get(otherTerritory).contains(territory))
+						continue;
+					for (final Area otherArea : territoryAreas.get(otherTerritory))
+					{
+						final Area testArea = new Area(scaledShape);
+						testArea.intersect(otherArea);
+						if (!testArea.isEmpty() && ConnectionFinder.sizeOfArea(testArea) > minOverlap)
+						{
+							thisTerritoryConnections.add(otherTerritory);
+						}
+						else if (!testArea.isEmpty())
+						{
+							
+						}
+					}
+				}
+				MapXMLHelper.s_territoyConnections.put(territory, thisTerritoryConnections);

```

Comment:
Again, this should be a 1st class method on MapXMLHelper, rather than allowing client code to "reach in" that class and modify its variables directly.